service/controller/app/v1/resource_set.go

```diff
@@ -131,6 +132,19 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		}
 	}
 
+	var kubeconfigFinalizerResource controller.Resource
```

Comment:
This name is a bit more, but I don't want to make it shorten since we already have `kubeconfig` library we imported in here.